util/pyclient/bft_msgs.py

```diff
@@ -53,13 +53,14 @@ def __init__(self, message):
 ReplyHeader = namedtuple('ReplyHeader', ['span_context_size', 'primary_id',
     'req_seq_num', 'length', 'rsi_length'])
 
-def pack_request(client_id, req_seq_num, read_only, timeout_milli, cid, msg, pre_process=False, span_context=b''):
+def pack_request(client_id, req_seq_num, read_only, timeout_milli, cid, msg, pre_process=False, span_context=b'', special_flags=None):
```

Comment:
This API addition is way too low-level. It requires the caller to know the hex value of the special flags. While that could be replaced with some global constant, it's better to instead not expose the flags, which are a low level detail of the protocol, and can be changed without needing to change the `pack_request` API.

Instead you should expose the behavior you want. In this case I think it would be better to continue with the pattern here and take another boolean that defaults to false : `wedge=false` or `stop_consensus=false`, and then sent the proper flag value in the function. This will also make the code at the call sites be more legible such that readers can understand the behavior without having to figure out what special flags are.

Reply:
fixed